lib/model-builder.js

```diff
@@ -596,14 +597,17 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
           } else {
             if (DataType === List) {
               this.__data[propertyName] = isClass(DataType) ?
-                new DataType(value, properties[propertyName].type, this.__data) :
-                DataType(value, properties[propertyName].type, this.__data);
+                new DataType(value, properties[propertyName].type, this) :
+                DataType(value, properties[propertyName].type, this);
             } else {
               // Assume the type constructor handles Constructor() call
               // If not, we should call new DataType(value).valueOf();
               this.__data[propertyName] = (value instanceof DataType) ?
                 value :
                 isClass(DataType) ? new DataType(value) : DataType(value);
+              if (value && typeof this.__data[propertyName] === 'object') { // we are dealing with an embedded model, apply parent
```

Comment:
I don't think this check is robust enough. What if the property is not an embedded model, but just an (anonymous) object? e.g.

```json
{
  "name": "MyModel",
  "base": "PersistedModel",
  "properties": {
    "name": {
      "type": "string",
      "required": true,
    },
    "customData": "object"
  }
}
```

Reply:
You are also right about this here. It is inconsistent that the check only verifies it is an object. I will check for model instance as well. (This is is done properly in **model.js** [here](https://github.com/strongloop/loopback-datasource-juggler/pull/1787/files#diff-a140b60b6a99a8c69f4b5d0d1f1f2bfaR352) )